h2/src/main/org/h2/mvstore/tx/TransactionStore.java

```diff
@@ -62,17 +63,38 @@
 
     private final DataType dataType;
 
-    private final BitSet openTransactions = new BitSet();
+    final AtomicReference<VersionedBitSet> openTransactions = new AtomicReference<>(new VersionedBitSet());
+
+    final AtomicReference<BitSet> committingTransactions = new AtomicReference<>(new BitSet());
 
     private boolean init;
```

Comment:
seeing the way this is acccessed sometimes inside a lock and sometimes not, perhaps it should be volatile?

Reply:
All those lock should go away very soon.
Volatile is not a replacement for Atomic... because... it’s not atomic.

Reply:
Sorry, did not realized at first, that you are talking about init and not commitingTransactions, i’ve added.
Yes, @katzyn is absolutely right on init variable usage